### PR TITLE
Revised Coin contract and Coin test contract

### PR DESCRIPTION
**Pull request by retotrinkler**
Moving coin and coin test to hydra examples

### CODE REVIEW

File: examples/native/coin/coin_contract.py [hunk 1]
```diff
@@ -1,84 +1,123 @@
-import hydrachain.native_contracts as nc
-from ethereum import utils
+import ethereum.utils as utils
+from hydrachain import native_contracts as nc
 import ethereum.slogging as slogging
-log = slogging.get_logger('nc.coin')
+from nc_utils import STATUS, FORBIDDEN, NOTFOUND, OK, ERROR, BADREQEST, INSUFFICIENTFUNDS
```
Comment (konradkonrad) [inline examples/native/coin/coin_contract.py]: import must be

``` python
from hydrachain.nc_utils ...
```

Comment (konradkonrad) [inline examples/native/coin/coin_contract.py]: `NOTFOUND`, `ERROR`, `BADREQUEST` are not in use. Don't import unused things :)

File: examples/native/coin/coin_contract.py [hunk 2]
```diff
@@ -1,84 +1,123 @@
...
-STATUS = 'uint16'
-FORBIDDEN = 403
-NOTFOUND = 404
-OK = 200
-ERROR = 500
-BADREQEST = 400
-INSUFFICIENTFUNDS = PAYMENTREQUIRED = 402
+class Transfer(nc.ABIEvent):
 
+    "Triggered when tokens are transferred."
```
Comment (konradkonrad) [inline examples/native/coin/coin_contract.py]: Please use proper docstrings!
https://github.com/amontalenti/elements-of-python-style#writing-good-docstrings

``` python
def get(url, qsargs=None, timeout=5.0):
    """Send an HTTP GET request.

    :param url: URL for the new request.
    :type url: str
    :param qsargs: Converted to query string arguments.
    :type qsargs: dict
    :param timeout: In seconds.
    :rtype: mymodule.Response
    """
return request('get', url, qsargs=qsargs, timeout=timeout)
```

Remember to use triple quotes.

File: examples/native/coin/coin_contract.py [hunk 3]
```diff
@@ -1,84 +1,123 @@
...
         ctx.owner = ctx.tx_origin
-        ctx.holdings[ctx.tx_origin] = 1000000
+        ctx.accounts[ctx.tx_origin] = _supply
+        ctx.supply = _supply
         return OK
 
-    @nc.constant
-    def creator(ctx, returns='address'):
-        return ctx.owner
+    # Standardized Contract API:
```
Comment (konradkonrad) [inline examples/native/coin/coin_contract.py]: should be a docstring

File: examples/native/coin/coin_contract.py [hunk 4]
```diff
@@ -1,84 +1,123 @@
...
+        if ctx.accounts[_from] >= _value and auth >= _value:
+            ctx.allowances[_from][ctx.msg_sender] -= _value
+            ctx.accounts[_from] -= _value
+            ctx.accounts[_to] += _value
+            ctx.Transfer(_from, _to, _value)
             return OK
         else:
             return INSUFFICIENTFUNDS
 
+    # Standardized Contract API:
```
Comment (konradkonrad) [inline examples/native/coin/coin_contract.py]: Should be docstring

File: examples/native/coin/coin_contract.py [hunk 5]
```diff
@@ -1,84 +1,123 @@
...
+    def approve(ctx, _spender='address', _value='uint256', returns=STATUS):
+        # approved[msg.sender].withdrawers[_addr]
+        ctx.allowances[ctx.msg_sender][_spender] += _value
+        ctx.Approval(ctx.msg_sender, _spender, _value)
+        return OK
 
-    @nc.constant
-    def numHolders(ctx, returns='uint32'):
-        return len(ctx.holdings)
+    # Standardized Contract API:
```
Comment (konradkonrad) [inline examples/native/coin/coin_contract.py]: Don't keep dead code. We're using git after all. If you **really** want to include this function as a stub for completeness sake, than raise a https://docs.python.org/2/library/exceptions.html#exceptions.NotImplementedError

File: examples/native/coin/coin_contract.py [hunk 6]
```diff
@@ -1,84 +1,123 @@
...
+    #   function unapprove(address _spender) returns (bool success)
+    # def unapprove(ctx, _spender='address', returns=STATUS):
+    #     ctx.allowances[ctx.msg_sender][_spender] = 0
+    #     return OK
+    # DEACTIVATED!
 
+    # Standardized Contract API:
```
Comment (konradkonrad) [inline examples/native/coin/coin_contract.py]: docstring

**Comment by konradkonrad** [comment]
Hey @howardpmarks,
this looks good, apart from the few commented things.
Could you work on the commented bits and update this branch please?

**Comment by retotrinkler** [comment]
sure

**Pull request merged**